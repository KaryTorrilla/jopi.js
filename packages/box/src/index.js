import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
