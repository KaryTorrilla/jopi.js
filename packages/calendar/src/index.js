import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
