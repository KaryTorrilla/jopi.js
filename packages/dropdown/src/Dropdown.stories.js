import React, { useState } from 'react'
import { Dropdown } from '.'
import { useFilterData } from '@oneloop/list'

export default {
  component: Dropdown,
  title: 'Dropdown',
}

const data = [
  {
    id: 22,
    content: {
      name: 'Agustin Iglesias',
      count: 10,
      icon: 'icon-alerta',
    },
  },
  {
    id: 23,
    content: {
      name: 'Josefina Golubicki',
      count: 43,
      icon: 'icon-adjuntar',
    },
  },
  {
    id: 8,
    content: {
      name: 'Wally',
      count: 45,
      icon: 'icon-agente',
    },
  },
]

/* ANTES DE SUBIR CHEQUEAR CON BRA, LA COMENTACION DE ESTOS DOS DROPDOWN
export const search = () =>
  React.createElement(() => {
    const [filteredData, setValue] = useFilterData(data, 'content')

    const onFilter = (e) => {
      setValue(e.target.value)
    }

    return (
      <Dropdown width={1 / 3}>
        <Dropdown.Button text='Placeholder'></Dropdown.Button>
        <Dropdown.Items>
          <Dropdown.Search placeholder='search something' onChange={onFilter} />
          {filteredData.map((user) => (
            <Dropdown.Item key={user.id} onClick={action('selected')}>
              {user.content.name}
            </Dropdown.Item>
          ))}
        </Dropdown.Items>
      </Dropdown>
    )
  })

export const select = () => (
  <Dropdown width={1 / 3}>
    <Dropdown.Button text='Opción elegida'></Dropdown.Button>
    <Dropdown.Items>
      {data.map((user) => (
        <Dropdown.Item key={user.id} onClick={action('selected')}>
          {user.content.name}
        </Dropdown.Item>
      ))}
    </Dropdown.Items>
  </Dropdown>
) */

export const DropdownDefault = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Opción elegida' filled={ list.length > 0 } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownDefaultDisabled = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDisabled' text='Opción elegida' filled={ list.length > 0 } disabled />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownDefaultWithIcon = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Opción elegida' icon='icon-ubicacion' filled={ list.length > 0 } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownOutlined = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownOutlined' text='Opción elegida' filled={ list.length > 0 } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownOutlinedWithIcon = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownOutlined' text='Opción elegida' icon='icon-ubicacion' filled={list.length > 0} />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownOutlinedWithIconDisabled = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownOutlinedDisabled' text='Opción elegida' icon='icon-ubicacion' filled={ list.length > 0 } disabled />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownTransparent = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownTransparent' text='Opción elegida' filled={ list.length > 0 } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownTransparentDisabled = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownTransparentDisabled' disabled text='Opción elegida' filled={ list.length > 0 } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownIcon = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 16}>
      <Dropdown.Button variant='dropdownIcon' isButtonIcon icon='icon-configuracion' filled={ list.length > 0 } variantSize='iconSmall' />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownIconDisabled = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 16}>
      <Dropdown.Button variant='dropdownIconDisabled' isButtonIcon disabled icon='icon-configuracion' filled={ list.length > 0 } variantSize='iconSmall' />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownButton = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 8}>
      <Dropdown.Button variant='dropdownButtonPrimary' text={'Label'} filled={ list.length > 0 } isArrowStatic></Dropdown.Button>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownButtonDisabled = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 8}>
      <Dropdown.Button variant='dropdownButtonPrimaryDisabled' disabled text={'Label'} filled={ list.length > 0 } isArrowStatic />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownColored = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownColored' text={'Label'} filled={ list.length > 0 } isArrowStatic />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesDefault = () => {
  const [value, setValue] = useState('Placeholder')

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text={value} filled={ value !== undefined } />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Default key={user.id} onClick={e => { setValue(user.content.name) }}>
            {user.content.name}
          </Dropdown.Default>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesMultiselect = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Seleccione' filled={ list.length > 0 }></Dropdown.Button>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} disabled={ user.id === 23 } >
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesMultiselectSearch = () => {
  const [list, setList] = useState([])
  const [filteredData, setValue] = useFilterData(data, 'content')

  const onFilter = (e) => {
    setValue(e.target.value)
  }

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }
  // AL ESCRIBIR EN EL SEARCH y cerrarlo se borra lo que se busco.
  // tengo que hacer que lo escrito en el input no se borre, porque lo necesito.
  // o bien perder el filtro, osea que aparezca todo de nuevo.
  // Creo que eso es mejor
  // Usar la opcion para que el filtro vuelva al total
  // DESPUES DE ESTO, QUEDA AGREGAR EL QUE TIENE EL ICONO; EL LUNES LO TERMINO SEGURO
  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Agente' filled={ list.length > 0 }></Dropdown.Button>
      <Dropdown.Items>
        <Dropdown.Search placeholder='Search' onChange={onFilter}/>
        {filteredData.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} disabled={ user.id === 23 } active={list.includes(user.id)}>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesIcon = () => {
  const [value, setValue] = useState('icon-configuracion')

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' icon={value} filled={ value !== undefined } isButtonIcon variantSize='iconSmall'/>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Icon key={user.id} onClick={e => { setValue(user.content.icon) }} icon={user.content.icon}>
            {user.content.name}
          </Dropdown.Icon>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

/** Other Sizes**/

export const DropdownDefaultSmall = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Opción elegida' filled={ list.length > 0 } variantSize = 'dropdownSizeSmall'/>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} variantSize = 'dropdownSizeSmall'>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownDefaultExtramall = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Opción elegida' filled={ list.length > 0 } variantSize = 'dropdownSizeExtraSmall'/>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} variantSize = 'dropdownSizeSmall'>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesIconSmall = () => {
  const [value, setValue] = useState('icon-configuracion')

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' icon={value} filled={value !== undefined} isButtonIcon variantSize='iconSmall'/>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Icon key={user.id} onClick={e => { setValue(user.content.icon) }} icon={user.content.icon} variantSize='dropdownSizeSmall'>
            {user.content.name}
          </Dropdown.Icon>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesIconExtraSmall = () => {
  const [value, setValue] = useState('icon-configuracion')

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' icon={value} filled={value !== undefined} isButtonIcon variantSize='iconExtraSmall'/>
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Icon key={user.id} onClick={e => { setValue(user.content.icon) }} icon={user.content.icon} variantSize='dropdownSizeExtraSmall'>
            {user.content.name}
          </Dropdown.Icon>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesMultiselectSmall = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Seleccione' filled={list.length > 0} variantSize = 'dropdownSizeSmall' />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} disabled={ user.id === 23 } variantSize = 'dropdownSizeSmall'>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}

export const DropdownStatesMultiselectExtraSmall = () => {
  const [list, setList] = useState([])

  const updateList = (value) => {
    if (!list.includes(value)) {
      setList([...list, value])
    } else {
      setList((list) => list.filter((id) => id !== value))
    }
  }

  return (
    <Dropdown width={1 / 4}>
      <Dropdown.Button variant='dropdownDefault' text='Seleccione' filled={list.length > 0} variantSize = 'dropdownSizeSmall' />
      <Dropdown.Items>
        {data.map((user) => (
          <Dropdown.Multiselect key={user.id} onClick={e => { updateList(user.id) }} disabled={ user.id === 23 } variantSize = 'dropdownSizeExtraSmall'>
            {user.content.name}
          </Dropdown.Multiselect>
        ))}
      </Dropdown.Items>
    </Dropdown>
  )
}
