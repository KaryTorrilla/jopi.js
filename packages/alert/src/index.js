import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
