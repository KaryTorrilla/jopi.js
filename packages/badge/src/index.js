import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
