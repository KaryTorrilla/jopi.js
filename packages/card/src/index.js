import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
