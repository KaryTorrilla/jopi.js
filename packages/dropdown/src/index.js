import React, { createContext, useContext } from 'react'
import { Box, Flex } from '@oneloop/box'
import { Button } from '@oneloop/button'
import { List } from '@oneloop/list'
import { useToggle } from '@oneloop/hooks'

const DropdownContext = createContext()

export const Dropdown = ({ children, ...props }) => {
  const [open, toggle] = useToggle(false)

  const value = React.useMemo(() => ({ open, toggle }), [open])

  return (
    <DropdownContext.Provider value={value}>
      <Box {...props} __css={{ display: 'inline-grid' }}>
        {children}
      </Box>
    </DropdownContext.Provider>
  )
}

const useDropdownContext = () => {
  const context = useContext(DropdownContext)
  if (!context) {
    throw new Error(
      `Dropdown compound components cannot be rendered outside the Dropdown component`
    )
  }
  return context
}

const DropdownButton = props => {
  const { toggle } = useDropdownContext()
  return (
    <Button
      {...props}
      onClick={toggle}
      sx={{
        textOverflow: 'ellipsis',
        overflow: 'hidden',
        whiteSpace: 'nowrap',
      }}
    />
  )
}

const DropdownList = props => {
  const { open } = useDropdownContext()
  return (
    open && (
      <Flex
        sx={{
          backgroundColor: 'neutral.0',
          boxShadow: 'active',
          borderRadius: 'default',
          margin: '5px 0 0',
        }}
      >
        <List
          {...props}
          __css={{
            flexGrow: 1,
            width: 0,
            mt: 0,
            mb: '5px',
          }}
        />
      </Flex>
    )
  )
}

Dropdown.Button = DropdownButton
Dropdown.Items = DropdownList
