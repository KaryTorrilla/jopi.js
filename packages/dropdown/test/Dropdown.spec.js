import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

import { Dropdown } from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('Dropdown', () => {
  test('default', () => {
    const component = shallow(
      <Dropdown width={1 / 4}>
        <Dropdown.Button>{'Opción elegida'}</Dropdown.Button>
        <Dropdown.Items>
          <Dropdown.Search />
          <Dropdown.Default>One</Dropdown.Default>
          <Dropdown.Default>Two</Dropdown.Default>
          <Dropdown.Default>Three</Dropdown.Default>
        </Dropdown.Items>
      </Dropdown>
    )
    expect(component.find(Dropdown.Button).exists).toBeTruthy()

    component.find(Dropdown.Button).simulate('click')
    component.find(Dropdown.Default).at(0).simulate('mouseover')
    component.find(Dropdown.Default).at(0).simulate('focus')
    component.find(Dropdown.Default).at(0).simulate('blur')

    expect(component).toMatchSnapshot()
  })

  test('DropdownMultiselect', () => {
    const tree = renderer
      .create(
        <Dropdown>
          <Dropdown.Button>{'Opción elegida'}</Dropdown.Button>
          <Dropdown.Items>
            <Dropdown.Search />
            <Dropdown.Multiselect disabled>One</Dropdown.Multiselect>
            <Dropdown.Multiselect>Two</Dropdown.Multiselect>
            <Dropdown.Multiselect>Three</Dropdown.Multiselect>
          </Dropdown.Items>
        </Dropdown>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('DropdownMultiselect', () => {
    const tree = renderer
      .create(
        <Dropdown>
          <Dropdown.Button icon='icon-configuracion' isButtonIcon />
          <Dropdown.Items>
            <Dropdown.Search />
            <Dropdown.Icon icon='icon-agente' disabled>One</Dropdown.Icon>
            <Dropdown.Icon icon='icon-agente'>Two</Dropdown.Icon>
            <Dropdown.Icon icon='icon-agente'>Three</Dropdown.Icon>
          </Dropdown.Items>
        </Dropdown>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('DropdownMultiselect', () => {
    const tree = renderer
      .create(
        <Dropdown>
          <Dropdown.Button icon='icon-configuracion' isButtonIcon />
          <Dropdown.Items>
            <Dropdown.Item icon='icon-agente' disabled>One</Dropdown.Item>
            <Dropdown.Item icon='icon-agente' variantSize='dropdownSizeNormal'>Two</Dropdown.Item>
            <Dropdown.Item icon='icon-agente'>Three</Dropdown.Item>
          </Dropdown.Items>
        </Dropdown>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('DropdownExpectError', () => {
    try {
      renderer
        .create(
          <>
            <Dropdown.Button icon='icon-configuracion' isButtonIcon />
            <Dropdown.Items>
              <Dropdown.Item icon='icon-agente' disabled>One</Dropdown.Item>
              <Dropdown.Item icon='icon-agente'>Two</Dropdown.Item>
              <Dropdown.Item icon='icon-agente'>Three</Dropdown.Item>
            </Dropdown.Items>
          </>
        )
        .toJSON()
    } catch (e) {
      expect(e.message).toBe('Dropdown compound components cannot be rendered outside the Dropdown component')
    }
  })
})
