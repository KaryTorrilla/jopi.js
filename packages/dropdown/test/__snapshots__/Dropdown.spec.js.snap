// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown default 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: inherit;
  font-family: Nunito Sans;
  font-weight: bold;
  font-size: 16px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  padding-left: 26px;
  padding-right: 26px;
  padding-top: 13px;
  padding-bottom: 13px;
  color: white;
  background-color: primary;
  border: 0;
  border-radius: button;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 48px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 16px;
  padding-right: 16px;
  text-align: start;
  font-size: 14px;
  line-height: 0;
  height: 36px;
  width: 236px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
      onClick={[Function]}
    >
      Opci√≥n elegida
    </button>
  </div>
</div>
`;
